garden-backend-service/src/models/failed_search_index.py

```diff
@@ -0,0 +1,18 @@
+from datetime import datetime
+
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.orm import Mapped, mapped_column
+from src.models.base import Base
+
+
+class FailedSearchIndexUpdate(Base):
+    __tablename__ = "failed_search_index_updates"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    doi: Mapped[str | None]
```

Comment:
I don't have strong feelings either way, I figured since we are essentially pulling the DOI out of the garden table, it should already be unique, and the insert logic should just update an existing record if it finds a duplicate DOI in this table. I am all for adding a unique constraint if that makes more sense/is clearer.

As far as possibly being None, I don't think there is a case where that will happen. I tend to make everything nullable until I have a solid reason not to. My thinking is that it is better to accept a partial record with some missing data than to reject it wholesale because of a missing field. However, in this case it doesn't make sense to have a failed update record that doesn't have an associated DOI, so I think making the DOI not null makes sense here.